apis/v1alpha2/validation/gateway.go

```diff
@@ -55,11 +83,13 @@ func validateGatewayListeners(listeners []gatewayv1a2.Listener, path *field.Path
 func validateListenerHostname(listeners []gatewayv1a2.Listener, path *field.Path) field.ErrorList {
 	var errs field.ErrorList
 	for i, h := range listeners {
-		// When unspecified, “”, or *, all hostnames are matched.
-		if h.Hostname == nil || (*h.Hostname == "" || *h.Hostname == "*") {
+		if h.Hostname == nil {
 			continue
 		}
 		hostname := string(*h.Hostname)
+		if isProtocolInSubset(h.Protocol, tcpUdpProtocols) {
+			errs = append(errs, field.Forbidden(path.Index(i).Child("hostname"), fmt.Sprintf("hostname should not be present for protocol type %v", h.Protocol)))
```

Comment:
Nit: make this message consistent with the other one

```suggestion
errs = append(errs, field.Forbidden(path.Index(i).Child("hostname"), fmt.Sprintf(" should  be empty for protocol  %v", h.Protocol)))
```

Reply:
Thanks Nice catch !! The new message looks more elegant